WordPress/Classes/Stores/StatsInsightsStore.swift

```diff
@@ -194,28 +196,26 @@ class StatsInsightsStore: QueryStore<InsightStoreState, InsightQuery> {
         processQueries()
     }
 
-    func persistToCoreData() {
-        guard
-            let siteID = SiteStatsInformation.sharedInstance.siteID,
-            let blog = Blog.lookup(withID: siteID, in: ContextManager.shared.mainContext) else {
-                return
+    func saveDataInCache() {
+        guard let siteID = SiteStatsInformation.sharedInstance.siteID else {
+            return
         }
+        func setValue<T: StatsInsightData>(_ value: T, _ record: StatsInsightsCache.Record) {
+            cache.setValue(value, record: record, siteID: siteID)
+        }
+        state.lastPostInsight.map { setValue($0, .lastPostInsight) }
```

Comment:
> Given both state and cache are not stored in memory, do you see value in having both? Theoretically, we could write and read straight from the cache without using state as an intermediary.

That's a good point. Yes, I think in this case, we could remove the state entirely, but this would've been a much larger change. I wanted to make it easy to review because the goal wasn't to refactor but to address the immediate issue.

Reply:
Okay, yes, let's not expand this PR with this additional refactoring!